src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -280,33 +274,78 @@ public function countByCustomerAndPaymentState(CustomerInterface $customer, $sta
         ;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function findBetweenDates(\DateTime $from, \DateTime $to, $state = null)
     {
-        $queryBuilder = $this->getCollectionQueryBuilderBetweenDates($from, $to, $state);
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        if (null !== $state) {
+            $queryBuilder
+                ->andWhere('o.state = :state')
+                ->setParameter('state', $state)
+            ;
+        }
 
         return $queryBuilder
+            ->andWhere($queryBuilder->expr()->isNotNull('o.completedAt'))
+            ->andWhere($queryBuilder->expr()->gte('o.createdAt', ':from'))
+            ->andWhere($queryBuilder->expr()->lte('o.createdAt', ':to'))
+            ->setParameter('from', $from)
+            ->setParameter('to', $to)
             ->getQuery()
             ->getResult()
         ;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function countBetweenDates(\DateTime $from, \DateTime $to, $state = null)
     {
-        $queryBuilder = $this->getCollectionQueryBuilderBetweenDates($from, $to, $state);
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        if (null !== $state) {
+            $queryBuilder
+                ->andWhere('o.state = :state')
+                ->setParameter('state', $state)
+            ;
+        }
 
         return $queryBuilder
             ->select('count(o.id)')
+            ->andWhere($queryBuilder->expr()->isNotNull('o.completedAt'))
+            ->andWhere($queryBuilder->expr()->gte('o.createdAt', ':from'))
+            ->andWhere($queryBuilder->expr()->lte('o.createdAt', ':to'))
+            ->setParameter('from', $from)
+            ->setParameter('to', $to)
             ->getQuery()
             ->getSingleScalarResult()
         ;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function revenueBetweenDates(\DateTime $from, \DateTime $to, $state = null)
     {
-        $queryBuilder = $this->getCollectionQueryBuilderBetweenDates($from, $to, $state);
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        if (null !== $state) {
+            $queryBuilder
+                ->andWhere('o.state = :state')
+                ->setParameter('state', $state)
+            ;
+        }
 
         return $queryBuilder
             ->select('sum(o.total)')
+            ->andWhere($queryBuilder->expr()->isNotNull('o.completedAt'))
+            ->andWhere($queryBuilder->expr()->gte('o.createdAt', ':from'))
+            ->andWhere($queryBuilder->expr()->lte('o.createdAt', ':to'))
+            ->setParameter('from', $from)
+            ->setParameter('to', $to)
```

Comment:
The same code used again in `revenueBetweenDates` and `countBetweenDates` (only select is different).